### DISCUSSION TITLE
Getting Cannot read property 'providers' of undefined when i run 'npm run dev'

### DISCUSSION THREAD

**Discussion by isaacssemugenyi** [Alexigbokwe/ExpressWebJs#33] [Q&A] [answered]
This is my first time working expressWebJs, i am trying to start the app but get the error below, how can solve this or what am I doing wrong? I have tsc version 4.4.3 installed, node version 14.18.0, npm version 6.14.15 and expresswebjs-ts-version version 3.0.0 

-Starting ExpressWebJs ...
(node:12112) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'providers' of undefined
    at Boot.autoLoadServiceProviders (C:\Users\dell\Documents\Projects\doctor-app\doctor-app-backend\node_modules\expresswebcorets\lib\App\index.js:138:29)
    at Boot.appInstance (C:\Users\dell\Documents\Projects\doctor-app\doctor-app-backend\node_modules\expresswebcorets\lib\App\index.js:88:20)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at Boot.listen (C:\Users\dell\Documents\Projects\doctor-app\doctor-app-backend\node_modules\expresswebcorets\lib\App\index.js:75:9)
    at Boot.start (C:\Users\dell\Documents\Projects\doctor-app\doctor-app-backend\node_modules\expresswebcorets\lib\App\index.js:72:9)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:12112) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:12112) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by Alexigbokwe** [comment]
Can I see your package.json file?

**Reply by isaacssemugenyi** [reply]
I am using windows and below is my package.json

{
  "name": "expresswebjs-ts-version",
  "version": "3.0.0",
  "description": "ExpressWebJs Framework Typescript Version",
  "main": "build/app.js",
  "scripts": {
    "test": "jest",
    "dev": "ts-node maker run-dev",
    "start": "node ./build/maker run-start",
    "build": "ts-node maker run-build"
  },
  "keywords": [
    "framework",
    "expresswebjs"
  ],
  "_moduleAliases": {
    "Elucidate": "./node_modules/expresswebcorets/lib"
  },
  "author": "Alex Igbokwe",
  "license": "ISC",
  "devDependencies": {
    "@types/express": "^4.17.11",
    "@types/jest": "^26.0.22",
    "@types/mongoose-paginate": "^5.0.8",
    "@types/node": "^14.14.17",
    "jest": "^26.6.3",
    "nodemon": "^2.0.6",
    "ts-jest": "^26.5.4",
    "ts-node": "^9.1.1",
    "tsconfig-paths": "^3.9.0",
    "tslint": "^6.1.3",
    "typescript": "^4.2.3",
    "typescript-transform-paths": "^2.2.3"
  },
  "dependencies": {
    "amqp-ts": "^1.8.0",
    "aws-sdk": "^2.933.0",
    "bcryptjs": "^2.4.3",
    "expressweb-scheduler-ts": "^1.0.2",
    "expresswebcorets": "^1.1.18",
    "expresswebjs-preset-ts": "^1.0.4",
    "log4js": "^6.3.0",
    "maker-console-ts": "^1.0.50",
    "module-alias": "^2.2.2",
    "mongoose": "^5.11.9",
    "mysql": "^2.18.1",
    "mysql2": "^2.2.5",
    "recursive-copy": "^2.0.13",
    "socket.io": "^3.1.0"
  }
}

**Reply by Alexigbokwe** [reply]
Have you ran npm install to install the packages

**Reply by isaacssemugenyi** [reply]
Yes I have, i just did it again the error is still the same

**Reply by Alexigbokwe** [reply]
Let’s have a session on google meet, and look at it together. Kindly drop your email

**Reply by isaacssemugenyi** [reply]
isaacpro01@gmail.com

**Reply by Alexigbokwe** [reply]
kindly check your mail for the invite

**Comment by Alexigbokwe** [comment, marked as answer]
You need to generate your .env file using this command
cp example.env .env

Then you set up your env values e.g your database values
Once that is done, for Windows users, you make use of the Git Bash terminal to run 'npm run dev'

Let me know if this is helpful

**Reply by isaacssemugenyi** [reply]
Thank you very much, this was very helpful
